test/test_public_bindings.py

```diff
@@ -272,6 +272,663 @@ def test_no_new_bindings(self):
+            "torch.utils.data.datapipes.iter.combining",
+            "torch.utils.data.datapipes.iter.filelister",
+            "torch.utils.data.datapipes.iter.fileopener",
+            "torch.utils.data.datapipes.iter.grouping",
+            "torch.utils.data.datapipes.iter.routeddecoder",
+            "torch.utils.data.datapipes.iter.selecting",
+            "torch.utils.data.datapipes.iter.streamreader",
+            "torch.utils.data.datapipes.iter.utils",
+            "torch.utils.data.datapipes.map",
+            "torch.utils.data.datapipes.map.callable",
+            "torch.utils.data.datapipes.map.combinatorics",
+            "torch.utils.data.datapipes.map.combining",
+            "torch.utils.data.datapipes.map.grouping",
+            "torch.utils.data.datapipes.map.utils",
+            "torch.utils.data.datapipes.utils",
+            "torch.utils.data.datapipes.utils.common",
+            "torch.utils.data.datapipes.utils.decoder",
+            "torch.utils.data.dataset",
+            "torch.utils.data.distributed",
+            "torch.utils.data.gen_pyi",
+            "torch.utils.data.graph",
+            "torch.utils.data.graph_settings",
+            "torch.utils.data.sampler",
+            "torch.utils.dlpack",
+            "torch.utils.ffi",
+            "torch.utils.file_baton",
+            "torch.utils.hipify",
+            "torch.utils.hipify.constants",
+            "torch.utils.hipify.cuda_to_hip_mappings",
+            "torch.utils.hipify.hipify_python",
+            "torch.utils.hipify.version",
+            "torch.utils.hooks",
+            "torch.utils.mkldnn",
+            "torch.utils.mobile_optimizer",
+            "torch.utils.model_dump",
+            "torch.utils.model_zoo",
+            "torch.utils.show_pickle",
+            "torch.utils.tensorboard",
+            "torch.utils.tensorboard.summary",
+            "torch.utils.tensorboard.writer",
+            "torch.utils.throughput_benchmark",
+            "torch.version"
+        }
+        for _, modname, ispkg in pkgutil.walk_packages(path=torch.__path__, prefix=torch.__name__ + '.'):
+            is_a_private_module = False
+            split_strs = modname.split('.')
+            for elem in split_strs:
+                if elem.startswith("_"):
+                    is_a_private_module = True
+                    break
+
+            if is_a_private_module:
+                continue
+
+            def module_starts_with_modname(elem, modname):
+                private_elem = elem.startswith('_')
+                obj = getattr(modname, elem)
+                elem_module = getattr(obj, '__module__') if hasattr(obj, '__module__') else None
+                elem_modname_starts_with_modname = elem_module is None or elem_module.startswith(modname)
+                if private_elem:
+                    assert elem_module is None or not elem_modname_starts_with_modname
+                else:
+                    assert elem_modname_starts_with_modname
```

Comment:
I don't think plain asserts are good here. These should be `self.assertTrue` at least. But I think a better thing would be to store all of the wrong APIs (in a list) and at the end, raise an error with a nice error message that contains all of the things that are wrong (it helps a lot when solving problems raised by such script as you can fix all of them in one go).

An example of that is here: https://github.com/pytorch/pytorch/blob/99db53eaa76bc76ac266b224962adb103603508d/docs/source/conf.py#L451-L458